Implement the common playback framework

penrods opened the pull request:
See the framework implementation in https://github.com/mycroftai/skill-playback-control

penrods committed:
Implement the common playback framework

See the framework implementation in https://github.com/mycroftai/skill-playback-control

Commit:
Merge branch '18.08' into feature/common-play-framework

penrods committed:
First pass at CommonPlaySkill

Implementing the prototype of the new mycroft.skills.core class CommonPlaySkill.
All skills which wish to handle "Play" type commands will simply derive from
this class, which will tie it in to the new Common Play framework, then implement
two abstract methods.  The CommonPlaySkill will do the rest.

Also did a little tidying-up of the skill itself, making it clearer what it was
doing.

forslund commented on __init__.py:
```diff
@@ -12,31 +12,178 @@
+        if result:
+            match = result[0]
+            level = result[1]
+            callback = result[3] if len(result) > 2 else None
```
should be result[2]

forslund commented on __init__.py:
```diff
@@ -12,31 +12,178 @@
+        super().__init__(name, bus)
+        self.audioservice = None
+
+    def initialize(self):
```
I changed this in the spotify skill to override bind instead so the user don't have to call super if they have an initialize method:

```python
    def bind(self, bus):
        if bus:
            super().bind(bus)
            self.audioservice = AudioService(self.bus)
            self.add_event('play:query', self.__handle_play_query)
            self.add_event('play:start', self.__handle_play_start)
```

forslund reviewed (commented):
I tested the CommonPlaySkill with spotify and found a i small typo and I replaced the initialize function with a bind function. So far everything works nicely and it cleaned up my code a bit further.

penrods committed:
Broke new API in common_play_skill.py

Broke out CommonPlaySkill as done in the Spotify skill To simplify synching while
working through the shape of the common base class in several skills.

penrods committed:
Switch to the mycroft-core version of CommonPlaySkill

penrods committed:
Modified signatures of CommonPlaySkill before merge
